packages/jest-runner/src/test_worker.js

```diff
@@ -69,33 +67,20 @@ const getResolver = (config, rawModuleMap) => {
   }
 };
 
-// Cannot be ESM export because of worker-farm
-module.exports = (
-  {config, globalConfig, path, rawModuleMap}: WorkerData,
-  callback: WorkerCallback,
-) => {
-  let parentExited = false;
-  const disconnectCallback = () => (parentExited = true);
-  const removeListener = () =>
-    process.removeListener('disconnect', disconnectCallback);
-  process.on('disconnect', disconnectCallback);
-
+export default async function({
+  config,
+  globalConfig,
+  path,
+  rawModuleMap,
+}: WorkerData): Promise<TestResult> {
   try {
-    runTest(path, globalConfig, config, getResolver(config, rawModuleMap)).then(
-      result => {
-        removeListener();
-        if (!parentExited) {
-          callback(null, result);
-        }
-      },
-      error => {
-        removeListener();
-        if (!parentExited) {
-          callback(formatError(error));
-        }
-      },
+    return await runTest(
+      path,
+      globalConfig,
+      config,
+      getResolver(config, rawModuleMap),
     );
-  } catch (error) {
-    callback(formatError(error));
+  } catch (err) {
+    throw formatError(err);
```

Comment:
any particular reason `formatError` return something not an actual error? I realise that is not new in this PR, it just struck me as odd